### PR TITLE
7395 duration consistency

### PR DESCRIPTION
**Pull request by arisah-ijeoma**

### CODE REVIEW

File: spec/models/timeline_entry_spec.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+describe TimelineEntry, type: :model do
+
+  context 'primary and secondary agents' do
+    let(:tle_1) { build(:project_step, scheduled_duration_days: 0) }
+    let(:tle_2) { build(:project_step) }
+
+    it 'raises error if duration is less than 1 day' do
+      expect(tle_1).not_to be_valid
+      expect(tle_1.errors[:scheduled_end_date].join).to match("Duration days can't be less than 1")
```
Comment (emitche) [inline spec/models/timeline_entry_spec.rb]: Match to message described above.

File: db/migrate/20171129145057_remove_default_from_duration.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class RemoveDefaultFromDuration < ActiveRecord::Migration
```
Comment (emitche) [inline db/migrate/20171129145057_remove_default_from_duration.rb]: This migration is an IrreversibleMigration.

File: app/models/project_step.rb [hunk 3]
```diff
@@ -431,4 +432,12 @@ def color_between(start, finish, fraction = 0.5, opacity = 1)
     r = start.each_with_index.map { |val, i| val + (finish[i] - val) * fraction }
     "hsla(#{r[0]}, #{r[1]}%, #{r[2]}%, #{opacity})"
   end
+
+  def scheduled_duration_invalid?
```
[the remark this answers is not in the document]
Comment (arisah-ijeoma) [reply app/models/project_step.rb]: 👍

[the remark this answers is not in the document]
Comment (arisah-ijeoma) [reply app/models/project_step.rb]: we accept nil as a value and the error should be thrown if duration is less than one

File: spec/models/project_step_spec.rb [hunk 4]
```diff
@@ -306,10 +306,10 @@
       end
 
       it 'updates scheduled_end_date' do
-        expect(step.scheduled_end_date).to eq parent_end + 4
+        expect(step.scheduled_end_date).to eq parent_end + 3
       end
 
-      it 'scheduled_start_date can be set to parent end + 1' do
```
[the remark this answers is not in the document]
Comment (arisah-ijeoma) [reply spec/models/project_step_spec.rb]: oops! in error

**Pull request merged**